### PR TITLE
Allow mobs with specific AI flag to move outside BSP tree.

### PR DESCRIPTION
**Pull request by skittles1**
Modified movement checking code to allow mobs with AI_MOVE_WALKTHROUGH_WALLS to move outside the BSP tree.

### CODE REVIEW

File: blakserv/ccode.c [hunk 1]
```diff
@@ -2640,6 +2643,13 @@ int C_CanMoveInRoomBSP(int object_id, local_var_type *local_vars,
 		return ret_val.int_val;
 	}
 
+	if (walk_outside_bsp.v.tag != TAG_INT)
+	{
+		bprintf("C_CanMoveInRoomBSP finecol dest can't use non int %i,%i\n",
```
Comment (cyberjunk) [inline blakserv/ccode.c]: The debugstring is incorrect --> "finecol"  :)

**Comment by skittles1** [comment]
Thanks :) this still has a bit of cleaning up to do, I wanted to get your opinion if this was the correct method to implement it, or if there's a better way have this flag work.

File: blakserv/roofile.c [hunk 2]
```diff
@@ -956,6 +956,14 @@ bool BSPGetStepTowards(room_type* Room, V2* S, V2* E, V2* P, unsigned int* Flags
    if (!Room || !S || !E || !P || !Flags)
       return false;
 
+   bool moveOutsideBSP = False;
+
+   if ((*Flags & 0x100000) == 0x100000)
```
Comment (cyberjunk) [inline blakserv/roofile.c]: What's this 0x100000 constant?

Comment (skittles1) [inline blakserv/roofile.c]: Placeholder for a constant - we were discussing whether to fix the AI_MOVE_WALKTHROUGH_WALLS at all in IRC, I verified this approach works but wasn't sure it was the best one so it's uploaded in a mostly complete state.

Comment (cyberjunk) [inline blakserv/roofile.c]: was more about not being a DEFINE constant ...

**Pull request merged**